File: 0303/solution.ts
class NumArray {
  private prefixSums: number[];
  constructor(nums: number[]) {
    this.prefixSums = [];
    let curr = 0;
    for (let num of nums) {
      curr += num;
      this.prefixSums.push(curr);
    }
  }

  sumRange(left: number, right: number): number {
    if (left === 0) {
      return this.prefixSums[right];
    } else {
      return this.prefixSums[right] - this.prefixSums[left - 1];
    }
  }
}

File: 0303/solution.java
class NumArray {
    int [] prefixes;
    public NumArray(int[] nums) {
        this.prefixes = new int[nums.length];
        for(int i = 0; i < nums.length; i++) {
            this.prefixes[i] = i == 0 ? nums[i] : this.prefixes[i - 1] + nums[i];
        }
    }
    
    public int sumRange(int left, int right) {
        return left == 0 ? this.prefixes[right] : this.prefixes[right] - this.prefixes[left - 1];
    }
}